lib/money/currency.rb

```diff
@@ -24,14 +25,28 @@ def find(currency_iso)
       def currencies
         @@currencies ||= Loader.load_currencies
       end
+
+      def normalize_currency(currency_code)
+        code = currency_code.to_s.downcase
+        return code if currencies.keys.include?(code)
+
+        code = currency_normalization_map[code.upcase]
```

Comment:
Why not have the normalization map in a downcased format to avoid this `upcase` and `downcase` shuffle